pytorch_lightning/trainer/trainer.py

```diff
@@ -1753,7 +1754,7 @@ def _parse_devices(
     def _log_api_event(event: str) -> None:
         torch._C._log_api_usage_once("lightning.trainer." + event)
 
-    def __init_profiler(self, profiler: Optional[Union[BaseProfiler, str]]) -> None:
+    def __init_profiler(self, profiler: Optional[Union[Profiler, BaseProfiler, str]]) -> None:
```

Comment:
this only needs to be a Profiler since BaseProfiler extends Profiler
```suggestion
    def __init_profiler(self, profiler: Optional[Union[Profiler, str]]) -> None:
```